File: 0747-largest-number-at-least-twice-of-others/747.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        index = -1 
        largest, second_largest = float('-inf'), float('-inf')

        for (i, num) in enumerate(nums):
            if num > largest:
                second_largest = largest
                largest = num
                index = i
            elif num > second_largest:
                second_largest = num

        return index if largest >= 2 * second_largest else - 1

File: 0747-largest-number-at-least-twice-of-others/747.rs
impl Solution {
    pub fn dominant_index(nums: Vec<i32>) -> i32 {
       use std::i32::MIN;
       let mut index: usize = 0;
       let (mut largest, mut second_largest): (i32, i32) = (MIN, MIN);

       for (i, &num) in nums.iter().enumerate() {
           if num > largest {
               second_largest = largest;
               largest = num;
               index = i;
           } else if num > second_largest {
               second_largest = num;
           }
       }

       if largest >= 2 * second_largest {
           index as i32
       }
       else {
           -1
       }
    }
}
